src/utils/permission/index.js

```diff
@@ -113,7 +114,7 @@
         ? PERMISSIONS.ANDROID.READ_EXTERNAL_STORAGE
         : PERMISSIONS.IOS.PHOTO_LIBRARY
     );
-    if (requestResult === RESULTS.GRANTED) {
+    if (requestResult === RESULTS.GRANTED || requestResult === RESULTS.LIMITED) {
       return {
         message: StringConstant.cameraPermissionGranted,
         success: true
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

> Note: This review was outside of the patch, so it was mapped to the patch with the greatest overlap. Original lines [117-121]

Similar to the previous comment, the check for `RESULTS.LIMITED` here is a good addition but should be handled appropriately based on the requirements of your application. If full gallery access is needed, you may want to prompt the user to change their permission settings.
```diff
-    if (requestResult === RESULTS.GRANTED || requestResult === RESULTS.LIMITED) {
+    if (requestResult === RESULTS.GRANTED) {
       return {
         message: StringConstant.cameraPermissionGranted,
         success: true
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->